File: Leetcode/48 - Rotate Image/solution.py
def rotate(self, matrix: List[List[int]]) -> None:
    n: int = len(matrix)
    # Step 1: Reverse the rows (top to bottom)
    for i in range(n // 2):
        matrix[i], matrix[n - i - 1] = matrix[n - i - 1], matrix[i]
    # Step 2: Transpose (swap elements across diagonal)
    for i in range(n):
        for j in range(i + 1, n):
            matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]

File: Leetcode/48 - Rotate Image/solution.cpp
void rotate(vector<vector<int>> &matrix)
{
    int n = matrix.size();
    // Step 1: Reverse the matrix vertically
    for (int i = 0; i < (n / 2); i++)
    {
        swap(matrix[i], matrix[n - i - 1]);
    }
    // Step 2: Transpose the matrix
    for (int i = 0; i < n; i++)
    {
        for (int j = (i + 1); j < n; j++)
        {
            swap(matrix[i][j], matrix[j][i]);
        }
    }
}